Bumps docs to latest MkDocs and forks 7.0 version

benarent opened the pull request:
This looks like a bit of beast of a PR, but in short it is

- Upgraded Gravity Docs to the latest(ish) MkDocs images 1.0.4
- Copied over 6.x -> 7.x ( no new content ) 
- Removed the https://gravitational.com/gravity/docs/ -> https://gravitational.com/gravity/docs/overview/ redirect. Overview is now on /docs/ similar to Teleport. 
- Theme has had a few small tweaks to work with latest MkDocs image

wadells committed:
Remove outdated instance documentation.

This content isn't presently helping us, and newer mkdocs complain
that it contains invalid links:

  WARNING -  Documentation file 'instance.md' contains a link to 'images/hc.png' which is not found in the documentation files.
  WARNING -  Documentation file 'instance.md' contains a link to '../deploy/wake.gravitational.io' which is not found in the documentation files.

I double checked that no other files were using these images.

benarent committed:
Bumps docs to latest MkDocs and forks 7.0 version

wadells reviewed (approved):
LGTM.  I investigated similar mkdocs work in my branch here https://github.com/wadells/gravity/commits/0e68d6cb4fd0891bae776a4bffa0d4c5b805c166 and all the changes I see agree.

Development wise, I'd encourage you to split the MkDocs related changes and the 7.0 copy into seperate commits.  Doing so makes the substantive changes much more easy to review (as I can filter the 7.x copy from review -- I know it is safe -- and focus on the more nuanced changes).  The more files a PR touches (e.g. if `vendor/` is touched) the more it helps reviewers to have several smaller commits.

If you're interested in trying to do this and would like some help, I'd be happy to do a call.  I've worked with git a ton and am quite familiar with rebasing.

benarent committed:
Following best practice

benarent commented on docs/Dockerfile:
```diff
@@ -13,10 +13,8 @@ RUN groupadd jenkins --gid=$GID -o && useradd jenkins --uid=$UID --gid=$GID --cr
-             apt-get -q -y update && \
-             apt-get install -q -y ruby-sass)
+RUN apt-get update -y
+RUN apt-get install -q -y ruby-sass parallel
```
Keep the tips coming, I've fixed it.

benarent committed:
Bring over Walts updates and clean up of docs

benarent committed:
Fix template

benarent committed:
Fix a ton of broken internal and external link using milv

benarent committed:
Backport Milv fixes to 6.x

benarent committed:
Fix 5.x milv doc issues

benarent committed:
Fixes version switcher

benarent committed:
Merge branch 'master' into benarent/docs/bump-version-7

wadells commented on docs/6.x/requirements.md:
```diff
@@ -133,17 +133,17 @@ These ports are used for Cluster operation and should be open between cluster no
 | 3080                    | TCP                | HTTPS                | ext         | controllers | Teleport Web  UI                         |
 | 3008-3012, 6060         | TCP                | HTTPS / gRPC         | all         | all         | Internal Gravity services                |
-| 32009                   | TCP                | HTTPS                | ext         | controllers | Gravity Cluster/OpsCenter Admin UI (ext) |
+| 32009                   | TCP                | HTTPS                | ext         | controllers | Gravity cluster.mdOpsCenter Admin UI (ext) |
```
Another find/replace error.

wadells commented on docs/6.x/workshops.md:
````diff
@@ -1018,7 +1018,7 @@ Let's dig a little deeper into this deployment:
 
 ```
-$ kubectl describe deployments/my-nginx
+$ kubectl describe deployments.mdmy-nginx
````
find/replace error

wadells commented on docs/6.x/workshops.md:
````diff
@@ -1198,7 +1198,7 @@ scaling up new ReplicaSet and scaling down old ReplicaSet:
 
 ```
-$ kubectl describe deployments/my-nginx
+$ kubectl describe deployments.mdmy-nginx
````
find/replace error

wadells commented on docs/6.x/workshops.md:
````diff
@@ -1290,7 +1290,7 @@ my-nginx-2896527177-cv3fd   0/1       ImagePullBackOff   0          55s
 
 ```
-$ kubectl describe deployments/my-nginx
+$ kubectl describe deployments.mdmy-nginx
````
find/replace error

wadells commented on docs/6.x/workshops.md:
````diff
@@ -1318,7 +1318,7 @@ Events:
 
 ```
-$ kubectl rollout history deployments/my-nginx
+$ kubectl rollout history deployments.mdmy-nginx
````
find/replace error

wadells commented on docs/6.x/workshops.md:
```diff
@@ -1495,10 +1495,10 @@ sudo docker push $(minikube ip):5000/mattermost-worker:2.1.0
 Mattermost's worker expects configuration to be mounted at:
 
-`/var/mattermost/config/config.json`
+`/var/mattermost/config.mdconfig.json`
```
find/replace error

wadells commented on docs/6.x/workshops.md:
````diff
@@ -1495,10 +1495,10 @@ sudo docker push $(minikube ip):5000/mattermost-worker:2.1.0
 
 ```
-$ cat mattermost/worker-config/config.json
+$ cat mattermost/worker-config.mdconfig.json
````
find/replace error

wadells commented on docs/6.x/workshops.md:
````diff
@@ -1648,7 +1648,7 @@ $ kubectl create -f mattermost/worker.yaml --record
 
 ```
-$ kubectl describe deployments/mattermost-worker
+$ kubectl describe deployments.mdmattermost-worker
````
find/replace error

wadells commented on docs/7.x/README.md:
```diff
@@ -2,7 +2,7 @@
 This Gravity documentation is public and is hosted on https://gravitational.com/gravity/docs
 It's based on [MkDocs](http://www.mkdocs.org/) with our own theme, similar to
-[Teleport docs](https://gravitational.com/teleport/docs/quickstart/).
+[Teleport docs](https://gravitational.com/teleport/docs/quickstart.md).
```
Do we want a `.md` suffix on this external link?

wadells commented on docs/7.x/changelog.md:
```diff
@@ -1801,7 +1801,7 @@ for details.
 #### Improvements
 
-* Add support for GCE cloud provider. See [Installing on Google Compute Engine](/installation/#installing-on-google-compute-engine)
+* Add support for GCE cloud provider. See [Installing on Google Compute Engine](installation/#installing-on-google-compute-engine)
```
Elsewhere you always added `.md` why not here?

wadells commented on docs/7.x/changelog.md:
```diff
@@ -1777,7 +1777,7 @@ install/upgrade.
 
 * Add multizone support for GCE clusters.
-* Update preflight checks to check iptables modules. See [requirements](/requirements#iptables-modules)
+* Update preflight checks to check iptables modules. See [requirements](requirements#iptables-modules)
```
Elsewhere you always added `.md` why not here?

wadells commented on docs/7.x/changelog.md:
```diff
@@ -1887,21 +1887,21 @@ install/upgrade.
 
 * Upgrade Kubernetes to `v1.9.6`.
-* Add support for more InfluxDB aggregate functions for use in [rollups](/monitoring/#rollups).
+* Add support for more InfluxDB aggregate functions for use in [rollups](monitoring/#rollups).
```
Elsewhere you always added `.md` why not here?

wadells commented on docs/7.x/changelog.md:
```diff
@@ -1820,15 +1820,15 @@ install/upgrade.
 * Fix an issue with user invite/reset CLI commands displaying incorrect URLs.
 * Fix an issue with duplicate progress entries during install.
-* Fix a few issues with communication between gravity agents and installer/cluster.
+* Fix a few issues with communication between gravity agents and installercluster.md.
```
find/replace error

wadells commented on docs/7.x/changelog.md:
```diff
@@ -2392,14 +2392,14 @@ transient error
 
 * New `tele create` command creates clusters via the OpsCenter.
-  See [Creating Remote Clusters](/manage/#creating-remote-clusters) for details.
+  See [Creating Remote Clusters](manage/#creating-remote-clusters) for details.
```
`.md`?

wadells commented on docs/7.x/changelog.md:
```diff
@@ -2566,7 +2566,7 @@ transient error
 #### Bugfixes
 
-* Downloaded apps/packages are upserted during installation to make retries on connection failures idempotent.
+* Downloaded appspack.mdages are upserted during installation to make retries on connection failures idempotent.
```
find/replace error

wadells commented on docs/7.x/changelog.md:
```diff
@@ -2727,8 +2727,8 @@ It contains a number of improvements and bugfixes.
 * Add ability to specify pod/service network CIDR range via UI and CLI
-* Add AWS IAM policy to the [docs](https://gravitational.com/gravity/docs/ver/5.x/pack/#aws-iam-policy)
-* Add runbook to recover the cluster after node failure [docs](https://gravitational.com/gravity/docs/ver/5.x/cluster/#recovering-a-node)
+* Add AWS IAM policy to the [docs](https://gravitational.com/gravity/docs/ver/5.xconfig.md#aws-iam-policy)
```
find/replace error

wadells commented on docs/7.x/changelog.md:
```diff
@@ -2727,8 +2727,8 @@ It contains a number of improvements and bugfixes.
-* Add AWS IAM policy to the [docs](https://gravitational.com/gravity/docs/ver/5.x/pack/#aws-iam-policy)
-* Add runbook to recover the cluster after node failure [docs](https://gravitational.com/gravity/docs/ver/5.x/cluster/#recovering-a-node)
+* Add AWS IAM policy to the [docs](https://gravitational.com/gravity/docs/ver/5.xconfig.md#aws-iam-policy)
+* Add runbook to recover the cluster after node failure [docs](https://gravitational.com/gravity/docs/ver/5.xcluster.md#recovering-a-node)
```
find/replace error

wadells commented on docs/7.x/cluster.md:
```diff
@@ -388,13 +388,13 @@ updating app from 0.0.1-alpha.1 to 0.0.1-alpha.2
 The operation has been created in manual mode.
 
-See https://gravitational.com/gravity/docs/cluster/#managing-operations for details on working with operation plan.
+See https://gravitational.com/gravity/docs/cluster.md#managing-operations for details on working with operation plan.
```
This seems unusual compared to other urls.  I believe mkdocs strips the `.md` before rendering to end users for other links?  Will that be done here too (being a fully qualified url instead of a relative url)?

webvictim replied:
This should ideally be a relative path for the purposes of working with documentation locally.

wadells replied:
In my patch drop just now, I didn't address any fully qualified urls, as I was focusing on `milv -ignore-external` output (which will be what our CI runs on each PR.  I'm willing to forgo external link cleanup (including this) for the time being - as this PR is becoming a bit of a slog, and I'd love to get it in.  I have some other work (for CI, to make sure we don't regress) based on top of it.

wadells commented on docs/7.x/config.md:
```diff
@@ -56,8 +56,8 @@ spec:
-  # kubelet configuration as described here: https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/
-  # and here: https://github.com/kubernetes/kubelet/blob/release-1.13/config/v1beta1/types.go#L62
+  # kubelet configuration as described here: https://kubernetes.io/docs/tasks/administer-cluster.mdkubelet-config-file/
+  # and here: https://github.com/kubernetes/kubelet/blob/release-1.13/config.mdv1beta1/types.go#L62
```
find/replace error

wadells commented on docs/7.x/glossary.md:
```diff
@@ -1,59 +1,61 @@
 Application Bundle is distributed via the Ops Center, it can optionally "phone home" for automating updates, remote monitoring and trouble shooting.
 
-## [Gravity Cluster or Cluster](/)
+After Gravity 6.0 OpsCenter, is refereed to as Gravity Hub.  
```
referred

wadells commented on docs/7.x/milv.config.yaml:
```diff
@@ -23,5 +23,30 @@ white-list-external:
 black-list:
-files:
\ No newline at end of file
+  - "./changelog.md"
```
I saw that changelog.md was blacklisted earlier, but I wasn't sure about that. https://gravitational.com/gravity/docs/changelog/ is a customer facing resource.  Is this blacklisting something else?

wadells commented on docs/7.x/requirements.md:
```diff
@@ -133,17 +133,17 @@ These ports are used for Cluster operation and should be open between cluster no
 | 3080                    | TCP                | HTTPS                | ext         | controllers | Teleport Web  UI                         |
 | 3008-3012, 6060         | TCP                | HTTPS / gRPC         | all         | all         | Internal Gravity services                |
-| 32009                   | TCP                | HTTPS                | ext         | controllers | Gravity Cluster/OpsCenter Admin UI (ext) |
+| 32009                   | TCP                | HTTPS                | ext         | controllers | Gravity cluster.mdOpsCenter Admin UI (ext) |
```
find/replace error

wadells commented on docs/7.x/testplan.md:
```diff
@@ -7,7 +7,7 @@
 
 - [ ] Install Hub in CLI mode.
-  - [ ] Verify can configure [OIDC connector](https://gravitational.com/gravity/docs/ver/6.x/config/#example-google-oidc-connector), for example:
+  - [ ] Verify can configure [OIDC connector](https://gravitational.com/gravity/docs/ver/6.x/config.md#example-google-oidc-connector), for example:
```
Should this be a relative link?

wadells replied:
Also, is this directory right place for the testplan to live?  Should it be blacklisted?

If you want to punt this, that is fine, as dealing with testplan stuff is not the point of this PR, and I don't think we need further scope creep.

webvictim replied:
Should indeed be relative

wadells commented on docs/7.x/workshops.md:
````diff
@@ -1018,7 +1018,7 @@ Let's dig a little deeper into this deployment:
 
 ```
-$ kubectl describe deployments/my-nginx
+$ kubectl describe deployments.mdmy-nginx
````
find/replace error

wadells commented on docs/7.x/workshops.md:
````diff
@@ -1198,7 +1198,7 @@ scaling up new ReplicaSet and scaling down old ReplicaSet:
 
 ```
-$ kubectl describe deployments/my-nginx
+$ kubectl describe deployments.mdmy-nginx
````
find/replace error

wadells commented on docs/7.x/workshops.md:
````diff
@@ -1318,7 +1318,7 @@ Events:
 
 ```
-$ kubectl rollout history deployments/my-nginx
+$ kubectl rollout history deployments.mdmy-nginx
````
find/replace error

wadells commented on docs/7.x/workshops.md:
````diff
@@ -1290,7 +1290,7 @@ my-nginx-2896527177-cv3fd   0/1       ImagePullBackOff   0          55s
 
 ```
-$ kubectl describe deployments/my-nginx
+$ kubectl describe deployments.mdmy-nginx
````
find/replace error

wadells commented on docs/7.x/workshops.md:
```diff
@@ -1495,10 +1495,10 @@ sudo docker push $(minikube ip):5000/mattermost-worker:2.1.0
 Mattermost's worker expects configuration to be mounted at:
 
-`/var/mattermost/config/config.json`
+`/var/mattermost/config.mdconfig.json`
```
find/replace error

wadells commented on docs/7.x/workshops.md:
````diff
@@ -1495,10 +1495,10 @@ sudo docker push $(minikube ip):5000/mattermost-worker:2.1.0
 
 ```
-$ cat mattermost/worker-config/config.json
+$ cat mattermost/worker-config.mdconfig.json
````
find/replace error

wadells commented on docs/7.x/workshops.md:
````diff
@@ -1648,7 +1648,7 @@ $ kubectl create -f mattermost/worker.yaml --record
 
 ```
-$ kubectl describe deployments/mattermost-worker
+$ kubectl describe deployments.mdmattermost-worker
````
find/replace error

wadells commented on docs/build.sh:
```diff
@@ -1,52 +1,37 @@
-#     * copy 5.x.yaml to 6.x.yaml and make changes there
-#     * edit theme/scripts.html and update the docVersions variable
+# IMPORTANT! To add a new version, say 8.1
+#     * copy 2.3.yaml to 8.1.yaml
```
2.3?

wadells reviewed (changes requested):
I didn't check all of 4.x and 5.x.  The issues common in both 6.x and 7.x (mostly find/replace errors) are probably present in those older versions too.

wadells commented on docs/6.x/faq.md:
```diff
@@ -156,3 +156,17 @@ the respective `tele build`, `gravity install` and `gravity upgrade`
+
+## Cluster Status
+
+Gravity provides the `gravity status` command to view [Cluster Status](/cluster/#cluster-status).
```
These two urls you picked up from master should probably get the `.md` and stripping of the leading `/` treatment you do elsewhere.

wadells commented on docs/build.sh:
```diff
@@ -1,52 +1,37 @@
+# cp index.html ../build/docs/index.html
 
-# create a symlink to the latest
+# create a symlink called 'latest' to the latest directory, like "3.1"
```
3.1?  Seems left over from teleport.

benarent committed:
Fixes find and replace errors.

benarent committed:
Merge branch 'benarent/docs/bump-version-7' of https://github.com/gravitational/gravity into benarent/docs/bump-version-7

benarent committed:
Use correct Markdown link format

benarent committed:
Other broken links

wadells reviewed (commented):
I rebased some of my build work on top of this branch and saw the following issues:

For 6.x and 7.x:

```
WARNING -  Documentation file 'changelog.md' contains a link to 'manage.md' which is not found in the documentation files. 
```

Manage.md is present for 4.x and 5.x, so no need to worry about that.

For 4.x and 5.x there are some double `.md.md` links.  I've flagged them inline.

wadells committed:
Remove trailing whitespace.

wadells committed:
Prune .md.md extensions.

These links were mangled during #1210 updates.

wadells committed:
Link across versions for past documentation.

4.x introduced `tele create` that have since been removed (or at least the
documentation for `tele create` has been purged).  Instead of trying to
link to contemporary docs from the release notes, link to the docs at
the time using a fully qualified (and thus external?) url.

wadells commented on docs/4.x/milv.config.yaml:
```diff
@@ -0,0 +1,52 @@
+white-list-internal:
+  - "mailto:sales@gravitational.com"
+  - "mailto:info@gravitational.com"
+  - "mailto:ben@gravitational.com"
```
Why is ben@gravitational.com whitelisted?

wadells replied:
Removed in e985d0e.

wadells committed:
Remove query strings from images.

Perhaps these were used for centering in the past, but they don't appear
to do anything today.  Removing them has no effect in `make run`.

wadells committed:
Remove glossary from docs.

This file hasn't been touched (except for cleanup) since 2017 or
earlier, and it is not linked from anywhere else in the documentation
(or the whole repo).

Furthermore a google query of "site:gravitational.com glossary" does not
even find this page.  It is truly hidden.

wadells committed:
Clean up all broken internal links.

As reported by milv.  Continuing on @benarent's good work from #1210.

wadells committed:
Clean up one-off milv ignores.

These didn't handle the necessary links with milv 0.6. Some were
superfluous and some needed further specification
(cluster.md#users-tokens instead of #users-tokens).

wadells committed:
Remove email addresses from milv safelist.

These emails never occur in our documentation, according to `make lint`
in the docs directory.  Best not to keep vestigial stuff around.

wadells committed:
Add changelog to milv checks.

This was excluded before due to a huge number of broken internal links,
and the fact that several of the release notes are release specific, and
have no corresponding docs in later releases.  That won't stop me
though.  No link left behind!

wadells committed:
Remove README.md from individual doc dirs.

We have a README.md at docs/README.md that is considerably more up to
date and useful.

wadells commented on docs/5.x/changelog.md:
```diff
@@ -1108,7 +1108,7 @@ for details.
 #### Improvements
 
-* Add support for GCE cloud provider. See [Installing on Google Compute Engine](/installation/#installing-on-google-compute-engine)
+* Add support for GCE cloud provider. See [Installing on Google Compute Engine](installation.md.md#installing-on-google-compute-engine)
```
Removed in 3bf199d.

wadells commented on docs/4.x/monitoring.md:
```diff
@@ -214,15 +214,15 @@ Following table shows the alerts Telekube ships with by default:
 
 Kapacitor will also trigger an email for each of the events listed above if stmp resource has been
-configured (see [configuration](/monitoring/#configuration) for details).
+configured (see [configuration](monitoring.md.md#configuration) for details).
```
Removed in 3bf199d.

wadells commented on docs/4.x/changelog.md:
```diff
@@ -241,14 +241,14 @@ install/upgrade.
 
 * Upgrade Kubernetes to `v1.9.6`.
-* Add support for more InfluxDB aggregate functions for use in [rollups](/monitoring/#rollups).
+* Add support for more InfluxDB aggregate functions for use in [rollups](monitoring.md.md#rollups).
```
Removed in 3bf199d.

wadells commented on docs/4.x/changelog.md:
```diff
@@ -155,7 +155,7 @@ for details.
 #### Improvements
 
-* Add support for GCE cloud provider. See [Installing on Google Compute Engine](/installation/#installing-on-google-compute-engine)
+* Add support for GCE cloud provider. See [Installing on Google Compute Engine](installation.md.md#installing-on-google-compute-engine)
```
Removed in 3bf199d.

wadells commented on docs/4.x/changelog.md:
```diff
@@ -131,7 +131,7 @@ install/upgrade.
 
 * Add multi-zone support for GCE clusters.
-* Update preflight checks to check iptables modules. See [requirements](/requirements#iptables-modules)
+* Update preflight checks to check iptables modules. See [requirements](requirements.md.md#iptables-modules)
```
Removed in 3bf199d.

wadells commented:
> I rebased some of my build work on top of this branch and saw the following issues:
> 
> For 6.x and 7.x:
> 
> ```
> WARNING -  Documentation file 'changelog.md' contains a link to 'manage.md' which is not found in the documentation files. 
> ```
> 
> Manage.md is present for 4.x and 5.x, so no need to worry about that.

All internal changelog links cleaned up in 7365447.

wadells committed:
Merge branch 'master' into benarent/docs/bump-version-7

wadells reviewed (approved):
I went ahead and made most of the changes I suggested, plus some other cleanup as caught by my `make lint` target, so I'm happy. Lets talk through tomorrow @benarent, and we can probably get this merged first thing.

benarent committed:
Merge branch 'master' into benarent/docs/bump-version-7

benarent commented:
@r0mant This PR includes a copy of 7.x docs, a large upgrade to the theme, and a bunch of link fixing. As a code owner can you please review / approve.

r0mant commented on docs/6.x.yaml:
```diff
@@ -1,43 +1,55 @@
     version: 6.x
+    product: gravity
+    gravity:
+        version: 6.3.6
```
What is this version here? Do we need to update this for every release?

benarent replied:
For Teleport we've started to use the specific binary version to make on-boarding smoother.  This makes it easier to update binary versions reference in the docs.. .e.g. https://gravitational.com/gravity/docs/quickstart/#getting-the-tools 

In Teleport we use it so we only have to update one place.
![image](https://user-images.githubusercontent.com/559288/76439584-d2f9cb80-6379-11ea-860a-b5606d110ee7.png)

benarent committed:
Merge branch 'master' into benarent/docs/bump-version-7
